src/gui/widgets/Fader.cpp

```diff
@@ -61,75 +62,50 @@ namespace lmms::gui
 
 SimpleTextFloat * Fader::s_textFloat = nullptr;
 
-Fader::Fader( FloatModel * _model, const QString & _name, QWidget * _parent ) :
-	QWidget( _parent ),
-	FloatModelView( _model, this ),
-	m_fPeakValue_L( 0.0 ),
-	m_fPeakValue_R( 0.0 ),
-	m_persistentPeak_L( 0.0 ),
-	m_persistentPeak_R( 0.0 ),
-	m_fMinPeak( 0.01f ),
-	m_fMaxPeak( 1.1 ),
-	m_back(embed::getIconPixmap("fader_background")),
-	m_leds(embed::getIconPixmap("fader_leds")),
+Fader::Fader(FloatModel* model, const QString& name, QWidget* parent) :
+	QWidget(parent),
+	FloatModelView(model, this),
+	m_fPeakValue_L(0.0),
+	m_fPeakValue_R(0.0),
+	m_persistentPeak_L(0.0),
+	m_persistentPeak_R(0.0),
+	m_fMinPeak(dbfsToAmp(-42)),
+	m_fMaxPeak(dbfsToAmp(9)),
 	m_knob(embed::getIconPixmap("fader_knob")),
-	m_levelsDisplayedInDBFS(false),
-	m_moveStartPoint( -1 ),
-	m_startValue( 0 ),
-	m_peakGreen( 0, 0, 0 ),
-	m_peakRed( 0, 0, 0 ),
-	m_peakYellow( 0, 0, 0 )
+	m_levelsDisplayedInDBFS(true),
+	m_moveStartPoint(-1),
+	m_startValue(0),
+	m_peakOk(10, 212, 92),
+	m_peakClip(193, 32, 56),
+	m_peakWarn(214, 236, 82),
+	m_unityMarker(63, 63, 63, 255),
+	m_renderUnityLine(true)
 {
 	if( s_textFloat == nullptr )
 	{
 		s_textFloat = new SimpleTextFloat;
 	}
 
-	init(_model, _name);
+	setWindowTitle( name );
+	setAttribute( Qt::WA_OpaquePaintEvent, false );
+	// For now resize the widget to the size of the previous background image "fader_background.png" as it was found in the classic and default theme
+	QSize minimumSize(23, 116);
+	setMinimumSize(minimumSize);
+	resize(minimumSize);
+	setSizePolicy(QSizePolicy::MinimumExpanding, QSizePolicy::MinimumExpanding);
+	setModel( model );
+	setHintText( "Volume:","%");
 
 	m_conversionFactor = 100.0;
 }
 
 
-Fader::Fader( FloatModel * model, const QString & name, QWidget * parent, QPixmap * back, QPixmap * leds, QPixmap * knob ) :
-	QWidget( parent ),
-	FloatModelView( model, this ),
-	m_fPeakValue_L( 0.0 ),
-	m_fPeakValue_R( 0.0 ),
-	m_persistentPeak_L( 0.0 ),
-	m_persistentPeak_R( 0.0 ),
-	m_fMinPeak( 0.01f ),
-	m_fMaxPeak( 1.1 ),
-	m_back(*back),
-	m_leds(*leds),
-	m_knob(*knob),
-	m_levelsDisplayedInDBFS(false),
-	m_moveStartPoint( -1 ),
-	m_startValue( 0 ),
-	m_peakGreen( 0, 0, 0 ),
-	m_peakRed( 0, 0, 0 )
+Fader::Fader(FloatModel* model, const QString& name, QWidget* parent, const QPixmap& knob) :
+	Fader(model, name, parent)
 {
-	if( s_textFloat == nullptr )
-	{
-		s_textFloat = new SimpleTextFloat;
-	}
-
-	init(model, name);
+	m_knob = knob;
```

Comment:
Couldn't this just be in the initialization list?

Reply:
This does not work because this constructor first calls the other one which initializes the knob to the default pixmap "fader_knob". This call then overrides the knob pixmap.